### PR TITLE
Generate Windows toolchain config

### PR DESCRIPTION
**Pull request by sunjayBhatia**
- To allow Linux and Windows toolchain configs to coexist, generate different platforms into 
  toolchains/configs/${OS_FAMILY}
- Bumps bazel-toolchains dependency to 3.1.2 for Windows changes
- Remove all toolchains so CI regenerates them
- Reorder pipeline jobs to be serial, eliminates race between Linux and
  Windows toolchain generation jobs pushing to envoy-build-tools master

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
cc @wrowe

**Comment by sunjayBhatia** [comment]
@lizan when you get a chance, we would appreciate a review of this PR

**Comment by sunjayBhatia** [comment]
Windows CI run should be fixed by https://github.com/bazelbuild/bazel-toolchains/pull/877

File: azure-pipelines.yml [hunk 1]
```diff
@@ -66,14 +101,13 @@ jobs:
       DOCKERHUB_PASSWORD: $(DockerPassword)
       GCP_SERVICE_ACCOUNT_KEY: $(GcpServiceAccountKey)
 
-- job: generate_toolchains_linux
-  dependsOn: "build_container_linux"
+- job: generate_toolchains_windows
```
Comment (lizan) [inline azure-pipelines.yml:104]: can this two generate_toolchains be templated/parameterized?

Comment (sunjayBhatia) [reply azure-pipelines.yml:81]: made an attempt in the latest commit

File: azure-pipelines.yml [hunk 2]
```diff
@@ -66,36 +79,22 @@ jobs:
...
-    displayName: Generate toolchains
-
+  - template: .azure-pipelines/generate_toolchains.yml
+    parameters:
+      bazeliskInstallCommandPrefix: ""
+      bazeliskInstallPath: "/usr/bin/bazel.exe"
+      bazeliskReleaseFilename: bazelisk-windows-amd64.exe
+      gcrImageName: "envoy-build-windows"
+      osFamily: "windows"
+  # Publish build artifacts in last step of the pipeline
```
Comment (sunjayBhatia) [inline azure-pipelines.yml:94]: @lizan not sure if this is the purpose of this artifact publish task, it seems like the intent is to run as the last step of the pipeline after everything has been committed, does this look right to you? (only doing it once after the last job)

Comment (sunjayBhatia) [reply azure-pipelines.yml:94]: @lizan also, can you clarify what these published artifacts are for/where are they used? is this task still necessary?

**Comment by sunjayBhatia** [comment]
The build finished, github seems to not have picked up the hook to change the status of the build here but it did pass: https://dev.azure.com/cncf/envoy/_build/results?buildId=40354&view=logs&j=3140c08e-0ceb-500c-a7fc-1cac300b6d64

File: azure-pipelines.yml (hunk 2 above)
Comment (lizan) [reply azure-pipelines.yml:94]: No it isn't necessary. I was used to use this tar to test generated toolchains from CI.

**Comment by sunjayBhatia** [comment]
Looks like this went green again but Github missed the success hook on the ubuntu container build job

**Comment by sunjayBhatia** [comment]
@lizan this should be ready to review again, we can merge master again if you want but there shouldn't be any conflicts. All the checks have passed

**Comment by lizan** [comment]
can you merge master?

**Comment by sunjayBhatia** [comment]
@lizan should be good to go with master merged

**Pull request merged**